jlm/hls/backend/rvsdg2rhls/distribute-constants.hpp

```diff
@@ -3,22 +3,42 @@
  * See COPYING for terms of redistribution.
  */
 
-#ifndef JLM_LIBJLM_SRC_BACKEND_HLS_RVSDG2RHLS_DISTRIBUTE_CONSTANTS_HPP
-#define JLM_LIBJLM_SRC_BACKEND_HLS_RVSDG2RHLS_DISTRIBUTE_CONSTANTS_HPP
+#ifndef JLM_HLS_BACKEND_RVSDG2RHLS_DISTRIBUTE_CONSTANTS_HPP
+#define JLM_HLS_BACKEND_RVSDG2RHLS_DISTRIBUTE_CONSTANTS_HPP
 
-#include <jlm/llvm/ir/operators/lambda.hpp>
-#include <jlm/llvm/ir/RvsdgModule.hpp>
-#include <jlm/rvsdg/region.hpp>
+#include <jlm/rvsdg/Transformation.hpp>
 
-namespace jlm
+namespace jlm::hls
 {
-namespace hls
+
+class ConstantDistribution final : rvsdg::Transformation
 {
-void
-distribute_constants(rvsdg::Region * region);
+public:
+  ~ConstantDistribution() noexcept override;
+
+  ConstantDistribution();
+
+  ConstantDistribution(const ConstantDistribution &) = delete;
+
+  ConstantDistribution(ConstantDistribution &&) = delete;
+
+  ConstantDistribution &
+  operator=(const ConstantDistribution &) = delete;
+
+  ConstantDistribution &
+  operator=(ConstantDistribution &&) = delete;
```

Comment:
Is it a bit overkill to delete move constructors and move assignment? This makes it impossible to put the transformations in a vector, for example

Reply:
Fixed.

Reply:
@phate It looks like you removed the lines that delete the copy constructor/assignment, not the move constructor/assignment. Since this class is completely stateless I guess we don't really need to delete either, but I think the move constructor is the most important to not delete

Reply:
Meh, sleeping on the wheel. Should be fixed now.